components/brave_wallet/rust/lib.rs

```diff
@@ -198,6 +215,32 @@ fn bytes_are_curve25519_point(bytes: &[u8]) -> bool {
     curve25519_dalek::edwards::CompressedEdwardsY::from_slice(bytes).decompress().is_some()
 }
 
+fn decode_bech32(input: &str) -> Bech32DecodeResult {
+  if let Ok((hrp, data, variant)) = bech32::decode(&input) {
+    if let Ok(as_u8) = Vec::<u8>::from_base32(&data) {
+      match variant {
+        bech32::Variant::Bech32m => {
+          return Bech32DecodeResult {
+            hrp: hrp,
+            data: as_u8,
+            variant : Bech32DecodeVariant::Bech32m };
+        },
+        bech32::Variant::Bech32 => {
+          return Bech32DecodeResult {
+            hrp: hrp,
+            data: as_u8,
+            variant : Bech32DecodeVariant::Bech32 };
+        }
+      }
+    }
+  };
+
+  return Bech32DecodeResult {
```

Comment:
The `return` keyword is redundant here; just leave the semicolon (`;`) off then end of the struct initializer.